pkg/util/hostname/drift.go

```diff
@@ -0,0 +1,161 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build !serverless
+
+package hostname
+
+import (
+	"context"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config/setup"
+	"github.com/DataDog/datadog-agent/pkg/telemetry"
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+)
+
+// driftService contains configuration for hostname drift detection
+type driftService struct {
+	initialDelay      time.Duration
+	recurringInterval time.Duration
+}
+
+// Default timing values
+const (
+	DefaultInitialDelay      = 20 * time.Minute
+	DefaultRecurringInterval = 6 * time.Hour
```

Comment:
I would make these constants private, as they are only meant to be use in this package 😄